posts/python-in-production-part-5-of-5.md

```diff
@@ -0,0 +1,250 @@
+
+In order not to overwhelm, I am going to assume that you already have a Remote Server up and Running with Docker installed, or at the very least, Docker installed locally.  If you do not have Docker running, you can find the installation instructions [here](https://docs.docker.com/engine/install/). I am also going to assume that you have kept your code up to date in the github repository that we set up in part 1.  What we will learn how to do:
+* Write a Jenkinsfile
+* Connect Github to Jenkins
+* Create a Build Pipeline
+
+# Create a Jenkinsfile
+
+When we attach our github repository to Jenkins, we will be able to tell it to *look* in our repository for instructions on how to build our code.  Those instructions live in a file named `Jenkinsfile`.  In the top level project directory `sample-python-project`, create a file called `Jenkinsfile`.
+```bash
+touch Jenkinsfile
+```
+
+Now we will populate it with our pipeline build instructions.  There are lots of different options and formats here, but we are going to keep it *as simple* as possible.  Open up the Jenkinsfile in your favorite text editor and write the following code.
+```groovy
+pipeline {
+    agent { label "built-in" }
+    stages {
+        stage('build') {
+            steps {
+                sh "bash build.sh"
+            }
+        }
+    }
+}
+
+```
+
+The code here is fairly straightforward to read.  If you were to find this file in the interwebs and attempted to read it you might come to these conclusions.  This is a pipeline file for Jenkins.  It consists of one stage named `build`, that runs one step.  The step is simply a shell command to run our `build.sh` file.  Remember the `build.sh` file we created in part 3, it will now be used inside of our Jenkins runner.  A Jenkins runner is just a linux vm.
+
+# Install Jenkins Plugins
+
+## Github Plugin
+
+Let's setup Jenkins to be able to pull our source code.  I will be using the Github plugin, because my source code is stored on Github.  Navigate to the left side bar and click on ***manage jenkins***.  Next, click on ***manage plugins***.  Search for `github` and install the `github` plugin.  Before you click install, we have one more plugin.
+
+## Pipeline Plugin
+
+When Jenkins pulls our source code, we want to be able to tell it to look for our Jenkinsfile.  This will allow us to update our build pipeline in our source code, without having to edit the jenkins job.  Search for `Pipeline` in the plugin manager.  Now you are ready to press install.
+
+# Create the Jenkins Job
+
+To create a Jenkins Job, click the `New Item` button on the left side bar. You should see the option `Pipeline`.  If you do not see this option, you may need to restart Jenkins.
```

Comment:
A screenshot of the UI and what you put in where would be useful for readability here